src/concrete/ml/quantization/base_quantized_op.py

```diff
@@ -933,39 +933,62 @@ def make_output_quant_parameters(
         """Round the input array to the specified number of bits.
 
         Args:
             x (Union[numpy.ndarray, fhe.tracing.Tracer]): The input array to be rounded.
             calibrate_rounding (bool): Whether to calibrate the rounding
                 (compute the lsbs_to_remove)
+            key (Optional[str]): The key to access the appropriate
+                lsbs_to_remove value in the dictionary.
 
         Returns:
             numpy.ndarray: The rounded array.
         """
 
-        # Rounding is applied only if specified by user
-        if self.rounding_threshold_bits is not None:
-            if calibrate_rounding:
-                assert_true(
-                    not isinstance(x, fhe.tracing.Tracer),
-                    "Can't compute lsbs_to_remove at compilation time.",
-                )
-                assert_true(
-                    self.lsbs_to_remove is None,
-                    "Rounding has already been calibrated.",
-                )
+        # Initialize lsbs_to_remove if not already set
+        if not hasattr(self, "lsbs_to_remove") or self.lsbs_to_remove is None:
+            self.lsbs_to_remove = {} if key else 0
 
-                current_n_bits_accumulator = compute_bits_precision(x)
-                self.lsbs_to_remove = current_n_bits_accumulator - self.rounding_threshold_bits
+        if self.rounding_threshold_bits is not None and calibrate_rounding:
+            # Compute new_value only when calibration is True
+            current_n_bits_accumulator = compute_bits_precision(x)
+            new_value = current_n_bits_accumulator - self.rounding_threshold_bits
 
-            # mypy
-            assert self.lsbs_to_remove is not None
+            assert_true(
+                not isinstance(x, fhe.tracing.Tracer),
+                "Can't compute lsbs_to_remove at compilation time.",
+            )
+
+            # Key-based logic (dictionary)
+            if key:
+                # mypy
+                assert isinstance(self.lsbs_to_remove, dict)
+                self.lsbs_to_remove[key] = max(self.lsbs_to_remove.get(key, 0), new_value)
 
-            # Apply rounding if needed
-            if self.lsbs_to_remove > 0:
-                x = fhe.round_bit_pattern(x, lsbs_to_remove=self.lsbs_to_remove)
+            # No key (integer logic)
+            else:
+                # mypy
+                assert isinstance(self.lsbs_to_remove, int)
+                self.lsbs_to_remove = max(self.lsbs_to_remove, new_value)
+
+        # Rounding logic
+        if isinstance(self.lsbs_to_remove, dict):
+            lsbs_value = self.lsbs_to_remove.get(key, 0) if key else 0
+        else:
+            lsbs_value = self.lsbs_to_remove
+
+        # Ensure lsbs_value is an integer
+        lsbs_value = int(lsbs_value)
+
+        if lsbs_value > 0:
+            x = fhe.round_bit_pattern(x, lsbs_to_remove=lsbs_value)
+        else:
+            self.lsbs_to_remove = None
 
         return x
```

Comment:
if `lsbs_value <= 0`, then x is not defined right ? I'm surprised mypy does not complain here, but if that's correct I would change the above code by something like : 
```
        if lsbs_value > 0:
            x = fhe.round_bit_pattern(x, lsbs_to_remove=lsbs_value)
            return x
        
        self.lsbs_to_remove = None
```

Reply:
Not sure what it brings but yes I can make that change if you want.

Reply:
ah my bad I didn't get that `x` was one of the method's parameter, no issues then (I thought it was undefined and thus would crash)